ulid_test.go

```diff
@@ -399,6 +399,48 @@ func TestEntropy(t *testing.T) {
 	}
 }
 
+type flakyEntropyReader struct {
+	data []byte
+	pos  int
+}
+
+func (r *flakyEntropyReader) Read(p []byte) (int, error) {
+	remaining := len(r.data) - r.pos
+	if remaining <= 0 {
+		return 0, io.EOF
+	}
+	n := len(p)
+	if n > 1 {
+		n /= 2
```

Comment:
Looking at this one more time made me remember https://golang.org/pkg/testing/iotest/#HalfReader which is doing the same thing. We should reuse that instead.

```
flakyEntropy := iotest.HalfReader(crand.Reader)
```

Reply:
Nice! I was not aware of the `iotest` package. Anyways, I rewrote the test completely using the `quick` package as the other tests do.